src/types.ts

```diff
@@ -1,4 +1,9 @@
 import { IParserResult } from '@stoplight/types';
-import { YAMLNode } from 'yaml-ast-parser';
+import { LoadOptions, YAMLNode } from 'yaml-ast-parser';
 
-export type YamlParserResult<T> = IParserResult<T, YAMLNode, number[]>;
+export interface IParseOptions extends LoadOptions {
+  json?: boolean; // if true, properties can be overridden, otherwise throws
+  mergeKeys?: boolean;
+}
+
+export type YamlParserResult<T> = IParserResult<T, YAMLNode, number[], IParseOptions>;
```

Comment:
Why does `IParseOption` be a generic parameter?

Reply:
Describes metadata.